### PR TITLE
fix:[JIRA:HCPSDKFIORIUIKIT-2076]SideBar-optimize the layout when the font becomes larger

### PR DESCRIPTION
**Pull request by KevinZK**
*optimize the layout of subTitle and accessoryIcon when the font becomes larger again. *make the icon dynamic

### CODE REVIEW

File: Sources/FioriSwiftUICore/_generated/ViewModels/API/SideBarListItem+API.generated.swift [hunk 1]
```diff
@@ -8,7 +8,9 @@ public struct SideBarListItem<Icon: View, Title: View, Subtitle: View, Accessory
 	@Environment(\.subtitleModifier) private var subtitleModifier
 	@Environment(\.accessoryIconModifier) private var accessoryIconModifier
 	@Environment(\.sideBarListItemConfigMode) var sideBarListItemConfigMode
-
+    @Environment(\.sizeCategory) var sizeCategory
+    @ScaledMetric var scale: CGFloat = 1
```
Comment (billzhou0223) [inline Sources/FioriSwiftUICore/_generated/ViewModels/API/SideBarListItem+API.generated.swift:12]: FYI, we utilize Sourcery to generate component APIs (All the files with `.generated` suffix are generated this way). This helps us ensure api consistency across components and avoid writing the same boilerplate code over and over again. Manually adding the changes in generated files will result in those changes being discarded when next time Sourcery runs. So we need to update `ModelDefinition.swift` file instead. This is the place where each component is defined and Sourcery will look for definitions there and generate API correspondingly. 

For more info, please refer to this section in readme. https://github.com/SAP/cloud-sdk-ios-fiori/blob/main/CONTRIBUTING.md#core-component-development

**Comment by billzhou0223** [comment]
Resolved by #408

File: Sources/FioriSwiftUICore/_generated/ViewModels/API/SideBarListItem+API.generated.swift (hunk 1 above)
Comment (KevinZK) [reply Sources/FioriSwiftUICore/_generated/ViewModels/API/SideBarListItem+API.generated.swift:12]: Thank you for the note

File: Sources/FioriSwiftUICore/Views/SideBarListItem+View.swift [hunk 2]
```diff
@@ -48,19 +48,48 @@ extension Fiori {
...
+                Spacer()
+                subtitle
+                    .font(.system(size: 17, weight: getFontWeight(), design: .default))
+                    .foregroundColor(.preferredColor(.tertiaryLabel))
+                accessoryIcon.foregroundColor(.preferredColor(.tertiaryLabel))
+                    .fixedSize()
+                    .frame(width: 22*scale, height: 22*scale)
+            }
+            .padding(EdgeInsets(top: 11, leading: 11, bottom: 11, trailing: 11))
+            .cornerRadius(8, antialiased: true)
```
[the remark this answers is not in the document]
Comment (billzhou0223) [reply Sources/FioriSwiftUICore/Views/SideBarListItem+View.swift:91]: I see. I will add it.

**Pull request closed without merging**